src/main/java/pl/allegro/tech/opel/LogicalOperatorExpressionNode.java

```diff
@@ -40,20 +40,21 @@ static LogicalOperatorExpressionNode orOperator(OpelNode left, OpelNode right, I
         try {
             return wrappedBody.get();
         } catch (Exception e) {
-            throw new OpelException("Error on evaluating left side of logical expression. " +
-                    "operator: " + logicalOperator + ", " +
-                    "left: " + left + ", class: " + left.getClass(), e);
+            String msg = String.format("Error on evaluating left side of logical expression. " +
+                                       "operator: '%s', left: '%s', class: '%s'",
+                    logicalOperator, left, left.getClass());
+            throw new OpelException(msg, e);
         }
     }
 
     private <T> T wrappingExceptionsWithOpelException(Object left, Object right, Supplier<T> wrappedBody) {
         try {
             return wrappedBody.get();
         } catch (Exception e) {
-            throw new OpelException("Error on evaluating logical expression. " +
-                    "operator: " + logicalOperator + ", " +
-                    "left: " + left + ", class: " + left.getClass() +
-                    "right: " + right + ", class: " + right.getClass(), e);
+            String msg = String.format("Error on evaluating logical expression. " +
+                                       "operator: '%s', left: '%s', class: '%s' right: '%s', class: '%s'",
+                    logicalOperator, left, left.getClass(), right, right.getClass());
+            throw new OpelException(msg, e);
```

Comment:
What do you think about moving message construction to specific exception?
`OpelLogicalExpressionException(cause: Exception, logicalOperator: Operator, left: Object, right: Object?)`?

Reply:
can do it, but it won't be reusable (both have different messages/args)

Reply:
done!